### DISCUSSION TITLE
Best way to trim space in paragraphs

### DISCUSSION THREAD

**Discussion by Tooning-FE-KimYongkuk** [ueberdosis/tiptap#3351] [General Questions]
```
<p ...>   a<strong>b</strong>   c    </p>
<p ...>     c</p>

=>

<p ...>a<strong>b</strong>   c</p>
<p ...>c</p>
```

**Comment by hanszoons** [comment]
I have the same issue, I can't seem to find a 'native' prosemirror solution either.

**Comment by ryanquinn3** [comment]
I've been scouring tiptap/prosemirror docs looking for a solution to this as well. We have a pretty simple schema that mainly consists of paragraphs and we want to remove leading/trailing whitespace from them once the user submits the document for saving to the server. 

I'm mostly looking for an easy way to do that trimming prior to submitting the json document to our server. We aren't rendering anything to html and most of the PM docs relating to whitespace relate to the HTML -> JSON parse step. We're exclusively using JSON in our code. 

Any help here would be a godsend 🙏

**Reply by nperez0111** [reply]
You could probably do a nodesBetween and trim any paragraph text yourself. There isn't really anything out of the box for these sorts of transformations. Although there should be

**Comment by davidstackio** [comment]
I just created an official feature request discussion for this: https://github.com/ueberdosis/tiptap/discussions/7066
